### ISSUE TITLE
[1.16.5] Missing UI textures in FD and JEI, potential conflicts with resource packs + related crashes

### ISSUE THREAD

**Issue by Whiskey-Sunshine** [vectorwing/FarmersDelight#276]
**Description**
When using JEI to look up recipes for FD (And sometimes by interacting with FD tile entities, especially the cooking pot) the UI will be blacked out or have the default missing texture purple and black, when using any additional resource packs. This includes the minecraft developer art pack, the xisumavoid vanilla tweaks packs (which only affect sounds and held item position in my pack), and packs that only adjust json models

After interacting with the missing UI, other textures may vanish. In my experience, this has included tiles or blocks that take up more than one block space (Beds, tall grass, tall flowers, to name a few), as well as item textures for dropped items and mobs.

Last, continuing to play after this point can potentially result in a crash, while attempting to re-order resource packs _will_ cause a crash.

Of note: This was not occurring until I installed FD as of 6/4/21, removing it does fix the issues mentioned above

**Steps to reproduce:**
Install FD, use a resource pack other than default, wait a while, check a recipe in JEI or interact with an FD UI of any kind. Afterwards, enjoy the potential missing textures and crashes.

**Expected behavior:**
UI appears as intended while using additional resource packs, does not cause textures to go invisible, does not cause a crash.

**Logs**
[crash-2021-06-05_17.42.25-client.txt](https://github.com/vectorwing/FarmersDelight/files/6603294/crash-2021-06-05_17.42.25-client.txt) Crash log regarding this
[2021-06-05-5.txt](https://github.com/vectorwing/FarmersDelight/files/6603295/2021-06-05-5.txt) Log from the same time

**Screenshots**
![image](https://user-images.githubusercontent.com/30344784/120907755-2072f000-c632-11eb-9707-280002ce778c.png)
Example screenshot of the issue

If there's anything you want or need me to do beyond this to help in figuring things out, don't hesitate to make demands.

**Mod list:**
Forge 36.1.2
Farmer's Delight 1.16.5-0.4.3
JEI 1.16.5-7.6.4.86

**Resource packs involved:**
Mizuno's 16 craft JE
scaryspider-1.16.2+
xisumavoid vanilla tweaks

**Shaders (Does not seem to be relevant, included just in case):**
Sildur's Vibrant Lite 1.281

**Full mod list:**
Abnormals Core
Abnormals Delight
Antique Atlas
Antique Waystones
AppleSkin
Aquaculture 2
Architectury
Artifacts
AutoRegLib
Backpacked
Backpacker
Balanced Enchanting
Berry Good
Better Foliage
BetterBurning
Bookshelf
BotanyPots
BotanyTrees
Bountiful
Caelus API
Carry On
CatJammies
Chunk Loaders
Citadel
Clay WorldGen
Cloth Config v4 API
CoFH Core
Colds: Beavers
Colds: Skunks
Colytra
Comforts
Cooking for Blockheads
CosmeticArmorReworked
CraftTweaker
Curio of Undying
Curios API
Diet
Direbats
Disenchanting
Dynamic Surroundings
Dynamic Trees
Dynamic Trees Plus
Dynamic Trees for Quark
Dynamic Trees for Traverse
Engineer's Decor
Ensorcellation
FTB GUI Library
Fairy Lights
Farmer's Delight
Farmer's Delight Compats
Farmer's Tea
Fast Furnace Minus Replacement
FastWorkbench Minus Replacement
Flat Bedrock
FlickerFix
Gauges and Switches
GeckoLib
Geolosys
Glassential
Ice and Fire
Immersive Cowardice
Immersive Engineering
Immersive Petroleum
Immersive Posts
JEI Professions
JEI Tweaker
Just Enough Items
Just Enough Resources
Kiwi
KleeSlabs
KubeJS
KubeJS Immersive Engineering
MattDahEpic Core
MineColonies
More Tea!
Mowzie's Mobs
Mystical World
Nature's Compass
No Increasing Repair Cost
NoLakes!
Ore Excavation
OreTweaker
Patchouli
Personality
Polymorph
Project: Vibrant Journeys
Quark
Rhino
Save My Stronghold!
Scarecrow's Territory
Serene Seasons
Simply Tea
Snow Under Trees
Snow! Real Magic!
Somnia Awoken
Spartan Shields
Spiders 2.0
Structurize
Stupid Horse Stand Still
SuperMartijn642's Config Lib
SuperMartijn642's Core Lib
SwingThroughGrass
Terra
TextDamageIndicators
The Endergetic Expansion
Tool Belt
Traverse: Reforged
Tumbleweed
Untitled Duck Mod
Upgrade Aquatic
Valhelsia Structures
Vanilla Degus
Waila
Waila Harvestability
Water Strainer
Waystones
Well-Behaved Mobs
Wyrmroost
YUNG's API
YUNG's Better Caves
YUNG's Better Mineshafts
YUNG's Better Strongholds
iChunUtil

**Comment by vectorwing** [comment]
Running through your game's logs; at first glance, it seems to be OptiFine tampering with GUI render methods. Something about it seems to be trying to access client render classes in the server thread, or something similar.

Please try running the following tests:
- Disable your shader and run through your steps again;
- Remove OptiFine from your pack and try again.

**Comment by Whiskey-Sunshine** [comment]
Right, so, I had to make a fresh instance because that pack will now crash with optifine removed. But, now, with the same mods without the shader, it does the same thing. Same mods with no shader and no optifine, the missing UI issue is fixed, but, uh...

![2021-06-05_19 25 35](https://user-images.githubusercontent.com/30344784/120908075-e9eaa480-c634-11eb-9a19-d9bf033a8e69.png) This happens now when adjusting resource packs. Relogging fixes it, until you move things again. Removing FD also fixes this particular weirdness.
 
[2021-06-05-1.log](https://github.com/vectorwing/FarmersDelight/files/6603400/2021-06-05-1.log)

Further update: Apparently removing, then re-adding Optifine and the shader again has fixed this. It's no longer log spamming, the UIs are functioning as intended, and there's no textures disappearing.

**Comment by vectorwing** [comment]
Oof, that's a weird bug. This new log also doesn't seem to report anything unusual. 😥

Optifine issues are usually very hard to troubleshoot since the mod is closed-source, so most of it ends up being guess-work. Though at least it seems to have resolved upon a reinstall, which is good. Sorry I couldn't assist further.

Have you tried an instance using only FD, JEI and Optifine?

**Comment by vectorwing** [comment]
Hi there, @Whiskey-Sunshine! Just checking in for an update. Did the issue resolve? If not, did you test the suggestion above?

**Comment by vectorwing** [comment]
I'll close it for now, since it has become stale. If the problem persists, feel free to reopen this issue and run the steps above. 👍

**Comment by Whiskey-Sunshine** [comment]
Apologies for the long delay, things got very busy for a bit.

I went over the steps you requested today, the problems kept appearing even in just FD + JEI + OF. It seems to have been an OptiFine issue entirely, as it did not occur without OF.

However, with some further testing, it seems that updating from OptiFine HD U G7 to HD U G8, the issues have been fixed entirely from their side. I will keep playing and testing for a bit to make sure it stays functional, and will reopen if it happens again.
